Revert encoding null values to 0x0602

staltz opened the pull request:
This PR is ready, but is pending on *design approval* from multiple stakeholders.

staltz committed:
Revert encoding null values to 0x0602

This reverts commit a7b209e3bf3c3f421092977f957d78f48ec50df8.

cryptix commented:
for future reference the discussion is here: https://github.com/ssb-ngi-pointer/ssb-binary-field-encodings-spec/issues/4#issuecomment-879746796

staltz commented:
This PR is now ready for review now that https://github.com/ssb-ngi-pointer/ssb-binary-field-encodings-spec/pull/6 has been merged

cryptix commented:
We should also change how the previous on message:1 is encoded, right?

staltz commented on internal/bencodeext/tanglepoint.go:
```diff
@@ -49,7 +41,7 @@ func (tp TanglePoint) MarshalBencode() ([]byte, error) {
 
 	if n := len(tp.Previous); n == 0 {
-		m["previous"] = zeroBendyMessage
+		m["previous"] = Null
```
@cryptix You mean here? It's supposed to be `[]byte{0x06, 0x02}`?

cryptix replied:
no, those are just the tangles.

I mean the encoding of previous on the first message on a feed.

cryptix committed:
change encoding of first message to Null

use 0x0602 (our bencode extentsion for null values) for the previous on
the first message, too

staltz commented:
Okay, you were much quicker to identify which files need to be changed, of course. Thanks for doing it! The only reason why I made this PR was to take some of the work off your shoulders after changing the spec. I hope I didn't bother.

cryptix commented:
No bother at all. Thanks for getting the ball rolling!

cryptix committed:
fix null encoding
